couchdbkit/designer/fs.py

```diff
@@ -102,9 +107,23 @@ def push(self, dbs, atomic=True, force=False):
                         db.put_attachment(doc, open(filepath, "r"),
                                             name=name)
 
-            logger.debug("%s/%s had been pushed from %s" % (db.uri,
-                self.docid, self.docdir))
+            if db_push:
+                logger.debug("%s/%s had been pushed from %s" % (db.uri,
+                    self.docid, self.docdir))
+            pushed |= db_push
+        return pushed
+
+    def _compare_with_current_version(self, db, doc):
+        """:returns: True if docs match otherwise False"""
+        try:
+            olddoc = db.open_doc(self._doc['_id'])
```

Comment:
Does this mean we're going to be doing two GETs per design doc? Probably fine / still a big improvement over an unnecessary write, but just wanted to make sure I was reading this right.

Reply:
That's correct. The other read happens here: https://github.com/dimagi/couchdbkit/blob/jsonobject/couchdbkit/designer/fs.py#L142 and (strangely) get's assigned to `self.olddoc`.

Technically I could just use that but it didn't feel right to make `_compare_with_current_version` rely on `self.olddoc` which gets re-assigned for each loop iteration (for each DB).